tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +722,19 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a hearbeat request ...
+                    if recordHeader.type == ContentType.heartbeat:
+                        try:
+                            heartbeat_request = Heartbeat().parse(p)
+                            #We answer with heartbeat response based on request
```

Comment:
the type of the received message will be different, and it needs to be handled differently:
 1. for requests - send reply, continue waiting for next messages
 2. for replies - call the callback, provide the parsed message to it, maybe the socket too (so that one callback can be reused between different sockets)